src/cpp/src/visual_language/phi3_vision/classes.cpp

```diff
@@ -734,7 +734,7 @@ ov::Tensor InputsEmbedderPhi3V::get_inputs_embeds(const std::string& image_promp
         }, chunk);
     }
 
-    if (!m_is_chat_conversation) {
+    if (!m_is_chat_conversation || m_use_full_chat_history) {
         m_tokens_per_images.clear();
```

Comment:
Should the full state of the embedder be reset instead? You can achieve that with `start_chat()` call from `VLMPipelineImpl`

Reply:
Great idea! Done.